### ISSUE TITLE
Add content & code license

### ISSUE THREAD

**Issue by keunes** [AntennaPod/antennapod.github.io#114]
**Short description**: pick & add a license for the website's content & code (aside MIT license of 'theme'), and mention it on the website.

**Location**: License page

**Why have this**: so (content & code) contributors know what they're committing to

**More info**:

**Comment by ByteHamster** [comment]
I think it's pretty hard to separate content/code and theme. I would say the whole website should be MIT

**Comment by keunes** [comment]
I'd say rather code/theme MIT and content CC-BY-NC

**Comment by comradekingu** [comment]
@keunes 
-NC is not a libre license, in turn running afoul of the terms in https://hosted.weblate.org/hosting/
"CC licensing" is therefore not a workable concept, because it doesn't qualify as really meaningful.

Going GPLv3+ on everything would at least work.
Is the theme made in-house, and if so does it have any utility for others?

**Comment by keunes** [comment]
>  -NC is not a libre license, in turn running afoul of the terms in https://hosted.weblate.org/hosting/

Right. I would've liked to avoid commercial copycats (which we have to deal with occasionally, unfortunately) using our content for their gain. But ok, indeed Weblate requires a 'OSI or FSF-approved license'. The FSF provides a list of licenses and [identified CC-BY-SA as a free license](https://www.gnu.org/licenses/license-list.html#ccbysa).

Let's use that for the content, then. @ByteHamster do we have your permission? (ByteHamster and I rewrote the content when preparing the new website.)

>  Is the theme made in-house, and if so does it have any utility for others?

The theme was made for us and released under MIT-license (but based largely on the K9 website, by the same creator). Whether it has value for others - I let others be the judge of that :)

**Comment by comradekingu** [comment]
If you want to run a straight copy and charge money for it that is up against "technically feasible", which is not made harder by having more licenses to read. That just makes it harder to try to go about it the right way.

I am pretty sure CC-BY-SA isn't compatible with GPLv3+ material, which would mean current app translations can't be used.
Wanting others to follow the terms of the GPLv3+ license in the first place means that should be understood to be forever unproblematic.
This is the premise of GPLv3+ _is_ GPLv3+, so why not that.

> 5. Conveying Modified Source Versions.
> 
> You may convey a work based on the Program, or the modifications to produce it from the Program, in the form of source code under the terms of section 4, provided that you also meet all of these conditions:
> 
>     a) The work must carry prominent notices stating that you modified it, and giving a relevant date.
>     b) The work must carry prominent notices stating that it is released under this License and any conditions added under section 7. This requirement modifies the requirement in section 4 to “keep intact all notices”.
>     c) You must license the entire work, as a whole, under this License to anyone who comes into possession of a copy. This License will therefore apply, along with any applicable section 7 additional terms, to the whole of the work, and all its parts, regardless of how they are packaged. This License gives no permission to license the work in any other way, but it does not invalidate such permission if you have separately received it.
>     d) If the work has interactive user interfaces, each must display Appropriate Legal Notices; however, if the Program has interactive interfaces that do not display Appropriate Legal Notices, your work need not make them do so.
> 
> A compilation of a covered work with other separate and independent works, which are not by their nature extensions of the covered work, and which are not combined with it such as to form a larger program, in or on a volume of a storage or distribution medium, is called an “aggregate” if the compilation and its resulting copyright are not used to limit the access or legal rights of the compilation's users beyond what the individual works permit. Inclusion of a covered work in an aggregate does not cause this License to apply to the other parts of the aggregate.
> 6. Conveying Non-Source Forms.
> 
> You may convey a covered work in object code form under the terms of sections 4 and 5, provided that you also convey the machine-readable Corresponding Source under the terms of this License, in one of these ways:
> 
>     a) Convey the object code in, or embodied in, a physical product (including a physical distribution medium), accompanied by the Corresponding Source fixed on a durable physical medium customarily used for software interchange.
>     b) Convey the object code in, or embodied in, a physical product (including a physical distribution medium), accompanied by a written offer, valid for at least three years and valid for as long as you offer spare parts or customer support for that product model, to give anyone who possesses the object code either (1) a copy of the Corresponding Source for all the software in the product that is covered by this License, on a durable physical medium customarily used for software interchange, for a price no more than your reasonable cost of physically performing this conveying of source, or (2) access to copy the Corresponding Source from a network server at no charge.
>     c) Convey individual copies of the object code with a copy of the written offer to provide the Corresponding Source. This alternative is allowed only occasionally and noncommercially, and only if you received the object code with such an offer, in accord with subsection 6b.
>     d) Convey the object code by offering access from a designated place (gratis or for a charge), and offer equivalent access to the Corresponding Source in the same way through the same place at no further charge. You need not require recipients to copy the Corresponding Source along with the object code. If the place to copy the object code is a network server, the Corresponding Source may be on a different server (operated by you or a third party) that supports equivalent copying facilities, provided you maintain clear directions next to the object code saying where to find the Corresponding Source. Regardless of what server hosts the Corresponding Source, you remain obligated to ensure that it is available for as long as needed to satisfy these requirements.
>     e) Convey the object code using peer-to-peer transmission, provided you inform other peers where the object code and Corresponding Source of the work are being offered to the general public at no charge under subsection 6d.
> 

AGPLv3+ adds network protection, and it means nothing can go from the website into the app without the license of the app changing.

The CC- "BY" is also something everyone sooner or later figured out wasn't helpful, but just created problems.
Creative Commons lends itself to the reputation of a community in which it doesn't belong, and manages to get it so wrong decades on.

I am not reading any of the CC licenses ever again to find out if it can work, because CC licensing didn't do anything productive.
For anything that happens to work, it is at that superfluous to other licenses.
In my eyes it is amounts to building-blocks to create a license that does a bunch of incompatible things.
They also know better than to pretend their CC0 license works in any meaningful sense for collaborative projects, but that too masquerades as functional.
Not only do they proliferate the marketplace of licenses, but they also put out 4 (!) versions of some of theirs, that respectively can be either international (intl) or not.
To top it all off Creative Commons loads Google Analytics on their website.

**Comment by keunes** [comment]
> If you want to run a straight copy and charge money for it that is up against "technically feasible", which is not made harder by having more licenses to read.

True. But that debate is out of scope as we can't use the -NC license anyway :)

> I am pretty sure CC-BY-SA isn't compatible with GPLv3+ material, which would mean current app translations can't be used.

That's fine. We are strictly talking here about the website. App translations are done elsewhere.

**Comment by comradekingu** [comment]
App vs content. in ranked order

GPL3+ and GPL3+ compatible, marketable. Everyone can deal with one license.
GPL3+ and AGPLv3+ network protection, slightly less marketable.
APLv3+ and AGPLv3+ network protection, marketable, makes it not possible to be on Google Play
GPLv3+ and CC anything not compatible in unknown ways, not marketable.

Since option 3 isn't feasibly an option, it takes some of the marketing out of 2, and I am not sure anyone wants to blindly copy just the website. Option 4 is inferior to all.

"Everything is GPLv3+" is vastly better for marketing, because it allows saying something "Use, verify, change, share; with all".

**Comment by comradekingu** [comment]
@keunes 
https://forum.antennapod.org/privacy doesn't specify a number, and it has changed a number of times since 2013.

Can you tell the difference between CC-BY-SA-4.0, 3.0, 2.5, 2.0 and 1.0?
What is the difference between INTL and the 18 (+?) different national variants?
Why do some people make it CC-BY-number-**plus**?

Maybe you know, but I don't, at all.

>Why have this: so (content & code) contributors know what they're committing to

**Comment by keunes** [comment]
@comradekingu I have no idea about the text of the forum privacy policy. It is default text that comes with Discourse, so you'll have [check with them](https://meta.discourse.org/) for an answer.
